public_html/wp-content/mu-plugins/blocks/assets/src/shared/item-select/option.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * External dependencies
+ */
+import { get }        from 'lodash';
+import createSelector from 'rememo';
+
+/**
+ * WordPress dependencies
+ */
+const { Dashicon } = wp.components;
+const { __ }       = wp.i18n;
+
+/**
+ * Internal dependencies
+ */
+import { AvatarImage }    from '../avatar';
+import { filterEntities } from '../../blocks-store';
+
+const buildOptionGroup = ( entityType, type, label, items ) => {
+	items = items.map( ( item ) => {
+		let parsedItem;
+
+		switch ( entityType ) {
+			case 'post':
+				parsedItem = {
+					label  : item.title.rendered.trim() || __( '(Untitled)', 'wordcamporg' ),
+					value  : item.id,
+					type   : type,
+				};
+
+				parsedItem.avatar = get( item, 'avatar_urls[\'24\']', '' );
+				parsedItem.image  = get( item, '_embedded[\'wp:featuredmedia\'].media_details.sizes.thumbnail.source_url', '' );
+				break;
+
+			case 'term':
+				parsedItem = {
+					label : item.name || __( '(Untitled)', 'wordcamporg' ),
+					value : item.id,
+					type  : type,
+					count : item.count,
+				};
+				break;
+		}
+
+		return parsedItem;
+	} );
+
+	return {
+		label   : label,
+		options : items,
+	};
+};
+
+export const buildOptions = createSelector(
+	( groups ) => {
+		const options = [];
+
+		groups.forEach( ( group ) => {
+			const { entityType, type, label, items } = group;
+			let orderby;
+
+			switch ( entityType ) {
+				case 'post':
+					orderby = 'title.rendered';
+					break;
+				case 'term':
+					orderby = 'name';
+					break;
+			}
+
+			if ( Array.isArray( items ) && items.length ) {
+				const sortedItems = filterEntities( items, { sort: orderby + '_asc' } );
+
+				options.push( buildOptionGroup( entityType, type, label, sortedItems ) );
+			}
+		} );
+
+		return options;
+	},
+	( groups ) => {
+		const references = [];
```

Comment:
This will always be a new object right? Which means result of previous call will never be equal to next call, which IIRC means memoization won't work. This can be verified by adding console log to both these functions and verifying that the calculation heavy function is not called often. 

Perhaps storing these calculated options in our data store could be a good idea, so that other blocks can also use them. Although I don't think this should be a blocker for this PR, even if options are not getting memoized right now because IMO we should focus on shipping this now.

Reply:
I tested this by adding a console log to the first function, which does the intensive data processing. It ran a couple of times, once for each time one of the data store requests finished and returned an array of entities. Once those were all loaded, I couldn't get it to run again, even by changing lots of the block's attributes and modifying the current selection in the select dropdown. So it seems like the memoization is actually working here.

Reply:
Ah ok :+1: . Resolving this conversation then. Thanks for this.